### ISSUE TITLE
Looking for help how to set up properly - apator162, MQTT

### ISSUE THREAD

**Issue by DoktorPrase** [wmbusmeters/wmbusmeters#1094]
### Type of request

Question

### HAOS version

2023.11.2

### wmbusmeters version

1.14.0

### Your message goes here

FIrst of all - thanks for this addon!
Second of all - I am not a linux guy, just started with HA, so i might be "stupid". Sorry

Now: rather fresh install, see config and log below:
mqtt section of configuration.yaml
```
mqtt:
  sensor:
    - state_topic: "wmbusmeters/KoupelnaTepla"
      json_attributes_topic: "wmbusmeters/KoupelnaTepla"
      unit_of_measurement: "m³"
      value_template: "{{ value_json.total_m3 }}"
      name: Water usage
      icon: "mdi:gauge"
      state_class: total_increasing
      device_class: water
    - state_topic: "wmbusmeters/KoupelnaStudena"
      json_attributes_topic: "wmbusmeters/KoupelnaStudena"
      unit_of_measurement: "m³"
      value_template: "{{ value_json.total_m3 }}"
      name: Water usage
      icon: "mdi:gauge"
      state_class: total_increasing
      device_class: water
    - state_topic: "wmbusmeters/WCTepla"
      json_attributes_topic: "wmbusmeters/WCTepla"
      unit_of_measurement: "m³"
      value_template: "{{ value_json.total_m3 }}"
      name: Water usage
      icon: "mdi:gauge"
      state_class: total_increasing
      device_class: water
    - state_topic: "wmbusmeters/WCStudena"
      json_attributes_topic: "wmbusmeters/WCStudena"
      unit_of_measurement: "m³"
      value_template: "{{ value_json.total_m3 }}"
      name: Water usage
      icon: "mdi:gauge"
      state_class: total_increasing
      device_class: water
    - state_topic: "wmbusmeters/KuchynTepla"
      json_attributes_topic: "wmbusmeters/Kuchyntepla"
      unit_of_measurement: "m³"
      value_template: "{{ value_json.total_m3 }}"
      name: Water usage
      icon: "mdi:gauge"
      state_class: total_increasing
      device_class: water
    - state_topic: "wmbusmeters/KuchynStudena"
      json_attributes_topic: "wmbusmeters/Kuchynstudena"
      unit_of_measurement: "m³"
      value_template: "{{ value_json.total_m3 }}"
      name: Water usage
      icon: "mdi:gauge"
      state_class: total_increasing
      device_class: water
```
wmbusmeters.conf:
```
device=auto:t1
donotprobe=/dev/ttyAMA0
format=json
logfile=/dev/stdout
loglevel=normal
logtelegrams=true
shell=shell=/usr/bin/mosquitto_pub -h localhost -t wmbusmeters/$METER_ID -m "$METER_JSON"
```
LOG:
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/nginx.sh
cont-init: info: /etc/cont-init.d/nginx.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun nginx (no readiness notification)
s6-rc: info: service legacy-services successfully started
[22:52:10] INFO: CONFIG_CONF ...
[22:52:10] INFO: {"device":"auto:t1","donotprobe":"/dev/ttyAMA0","format":"json","logfile":"/dev/stdout","loglevel":"normal","logtelegrams":true,"shell":"shell=/usr/bin/mosquitto_pub -h localhost -t wmbusmeters/$METER_ID -m \"$METER_JSON\""}
[22:52:10] INFO: CONFIG_METERS ...
[22:52:10] INFO: [{"driver":"apator162","id":"04721044","key":"00000000000000000000000000000000","name":"KoupelnaStudena"},{"driver":"apator162","id":"04375739","key":"00000000000000000000000000000000","name":"KoupelnaTepla"},{"driver":"apator162","id":"04720835","key":"00000000000000000000000000000000","name":"WCStudena"},{"driver":"apator162","id":"04377717","key":"00000000000000000000000000000000","name":"WCTepla"},{"driver":"apator162","id":"04705260","key":"00000000000000000000000000000000","name":"KuchynStudena"},{"driver":"apator162","id":"04387905","key":"00000000000000000000000000000000","name":"KuchynTepla"}]
[22:52:10] INFO: Syncing wmbusmeters configuration ...
[22:52:11] INFO: Registering meters ...
[22:52:11] INFO: Adding meter-0001 ...
[22:52:11] INFO: Adding meter-0002 ...
[22:52:11] INFO: Adding meter-0003 ...
[22:52:11] INFO: Adding meter-0004 ...
[22:52:12] INFO: Adding meter-0005 ...
[22:52:12] INFO: Adding meter-0006 ...
[22:52:12] INFO: Generating MQTT configuration ... 
[22:52:13] INFO: Broker core-mosquitto will be used.
[22:52:13] INFO: 
MQTT Discovery ...
tar: wmbusmeters-master/ha-addon/mqtt_discovery: not found in archive
[22:52:15] INFO:  Adding meter: KoupelnaStudena ...
[22:52:15] INFO:   File /config/wmbusmeters/etc/mqtt_discovery/apator162.json not found.
[22:52:15] INFO:  Adding meter: KoupelnaTepla ...
[22:52:15] INFO:   File /config/wmbusmeters/etc/mqtt_discovery/apator162.json not found.
[22:52:16] INFO:  Adding meter: WCStudena ...
[22:52:16] INFO:   File /config/wmbusmeters/etc/mqtt_discovery/apator162.json not found.
[22:52:16] INFO:  Adding meter: WCTepla ...
[22:52:16] INFO:   File /config/wmbusmeters/etc/mqtt_discovery/apator162.json not found.
[22:52:17] INFO:  Adding meter: KuchynStudena ...
[22:52:17] INFO:   File /config/wmbusmeters/etc/mqtt_discovery/apator162.json not found.
[22:52:17] INFO:  Adding meter: KuchynTepla ...
[22:52:17] INFO:   File /config/wmbusmeters/etc/mqtt_discovery/apator162.json not found.
[22:52:17] INFO: MQTT Discovery cleanup...
[22:52:18] INFO: Starting web configuration service.
[22:52:18] INFO: Running wmbusmeters ...
Started auto im871a[00104164] on /dev/ttyUSB0 listening on t1
[22:52:19] INFO: Starting Nginx...
2023/11/15 22:52:19 [notice] 83#83: using the "epoll" event method
2023/11/15 22:52:19 [notice] 83#83: nginx/1.24.0
2023/11/15 22:52:19 [notice] 83#83: OS: Linux 6.1.21-v8
2023/11/15 22:52:19 [notice] 83#83: getrlimit(RLIMIT_NOFILE): 1073741816:1073741816
2023/11/15 22:52:19 [notice] 83#83: start worker processes
2023/11/15 22:52:19 [notice] 83#83: start worker process 357
telegram=|6E4401066052700405077AD50060852F2F_0FB2AF6B9718020043A401830008DA8188CB0100310410882100007B018D1F0000271C0000051800005716000012140000C4110000780E0000960B0000440A0000090800003E05000052020000A037349B04FFFFFFFFFFFFFFFFFFFFD5A1|+13
/bin/sh: -h: not found
(shell) /bin/sh exited with non-zero return code: 127
172.30.32.2 - - [15/Nov/2023:22:52:34 +0100] "GET / HTTP/1.1" 200 1298 "http://192.168.1.215:8123/c8a990ad_wmbusmeters-ha-addon" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:22:52:34 +0100] "GET /bootstrap.min.css HTTP/1.1" 200 221227 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:22:52:34 +0100] "GET /bootstrap.bundle.min.js HTTP/1.1" 200 80762 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:22:52:34 +0100] "GET /main.js HTTP/1.1" 200 9980 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:22:52:34 +0100] "GET /wmbusmeters.png HTTP/1.1" 200 55338 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:22:52:35 +0100] "GET /get_json HTTP/1.1" 200 938 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
```

First suspicious thing:
```
[22:52:13] INFO: 
MQTT Discovery ...
tar: wmbusmeters-master/ha-addon/mqtt_discovery: not found in archive
```

Second suspicious thing:
```
[22:52:15] INFO:  Adding meter: KoupelnaStudena ...
[22:52:15] INFO:   File /config/wmbusmeters/etc/mqtt_discovery/apator162.json not found.
```
Third suspicious thing:
```
/bin/sh: -h: not found
(shell) /bin/sh exited with non-zero return code: 127
```

If I disable MQTT discovery, thing seem better (But what am I missing then?)
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/nginx.sh
cont-init: info: /etc/cont-init.d/nginx.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun nginx (no readiness notification)
s6-rc: info: service legacy-services successfully started
[23:12:53] INFO: CONFIG_CONF ...
[23:12:53] INFO: {"device":"auto:t1","donotprobe":"/dev/ttyAMA0","format":"json","logfile":"/dev/stdout","loglevel":"normal","logtelegrams":true,"shell":"shell=/usr/bin/mosquitto_pub -h localhost -t wmbusmeters/$METER_ID -m \"$METER_JSON\""}
[23:12:53] INFO: CONFIG_METERS ...
[23:12:53] INFO: [{"driver":"apator162","id":"04721044","key":"00000000000000000000000000000000","name":"KoupelnaStudena"},{"driver":"apator162","id":"04375739","key":"00000000000000000000000000000000","name":"KoupelnaTepla"},{"driver":"apator162","id":"04720835","key":"00000000000000000000000000000000","name":"WCStudena"},{"driver":"apator162","id":"04377717","key":"00000000000000000000000000000000","name":"WCTepla"},{"driver":"apator162","id":"04705260","key":"00000000000000000000000000000000","name":"KuchynStudena"},{"driver":"apator162","id":"04387905","key":"00000000000000000000000000000000","name":"KuchynTepla"}]
[23:12:53] INFO: Syncing wmbusmeters configuration ...
[23:12:54] INFO: Registering meters ...
[23:12:54] INFO: Adding meter-0001 ...
[23:12:54] INFO: Adding meter-0002 ...
[23:12:54] INFO: Adding meter-0003 ...
[23:12:54] INFO: Adding meter-0004 ...
[23:12:55] INFO: Adding meter-0005 ...
[23:12:55] INFO: Adding meter-0006 ...
[23:12:55] INFO: Generating MQTT configuration ... 
[23:12:56] INFO: Broker core-mosquitto will be used.
[23:12:56] INFO: MQTT Discovery cleanup...
[23:12:57] INFO: Starting web configuration service.
[23:12:57] INFO: Running wmbusmeters ...
Started auto im871a[00104164] on /dev/ttyUSB0 listening on t1
[23:12:59] INFO: Starting Nginx...
2023/11/15 23:12:59 [notice] 83#83: using the "epoll" event method
2023/11/15 23:12:59 [notice] 83#83: nginx/1.24.0
2023/11/15 23:12:59 [notice] 83#83: OS: Linux 6.1.21-v8
2023/11/15 23:12:59 [notice] 83#83: getrlimit(RLIMIT_NOFILE): 1073741816:1073741816
2023/11/15 23:12:59 [notice] 83#83: start worker processes
2023/11/15 23:12:59 [notice] 83#83: start worker process 256
172.30.32.2 - - [15/Nov/2023:23:13:29 +0100] "GET / HTTP/1.1" 200 1288 "http://192.168.1.215:8123/c8a990ad_wmbusmeters-ha-addon" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:23:13:29 +0100] "GET /bootstrap.min.css HTTP/1.1" 200 221221 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:23:13:29 +0100] "GET /main.js HTTP/1.1" 200 9981 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:23:13:29 +0100] "GET /bootstrap.bundle.min.js HTTP/1.1" 200 80762 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:23:13:29 +0100] "GET /wmbusmeters.png HTTP/1.1" 200 55338 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
172.30.32.2 - - [15/Nov/2023:23:13:29 +0100] "GET /get_json HTTP/1.1" 200 939 "http://192.168.1.215:8123/api/hassio_ingress/lNBH2KwcyyKAl1KCl0ukhufvx5bbxFGi2I30kKgbmrw/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
telegram=|6E4401063957370405077AC90060852F2F_0FC0AF6B97080200430C02830018D65C88CB01B8291210225100007B01F94C000065450000AC3D0000923A0000DE350000A5300000F6280000C5200000481C000045170000570F000040070000A0CF259304FFFFFFFFFFFFFFFFFFFFB9C1|+148
/bin/sh: -h: not found
(shell) /bin/sh exited with non-zero return code: 127
telegram=|6E4401060579380405077AD10060852F2F_0FC0AF6B97080200430C02830018D66C9CF80399D21310781900007B01C6170000CB140000501200009A11000060100000C10E0000300C0000B2090000C208000020070000C80400000B020000A0C9239304FFFFFFFFFFFFFFFFFFFFE98B|+158
/bin/sh: -h: not found
(shell) /bin/sh exited with non-zero return code: 127
telegram=|6E4401064410720405077A6D0060852F2F_0FCFAF6B9708020043A301830008DA2288CB01F92E0410609300007B01B38C0000EF7E00004C710000BA690000315F0000D755000025480000E83900005931000005270000281A0000990A0000A034EF9A04FFFFFFFFFFFFFFFFFFFF6A28|+189
(shell) /bin/sh exited with non-zero return code: 127
/bin/sh: -h: not found
telegram=|6E4401063508720405077A2F0060852F2F_0FD4AF6B9708000043A301830008DA8A68CB01982B0410AB6300007B019E5E00000B5600003F4D00009A48000025420000BA3A00006F30000026270000C82100000B1A0000AD10000045070000A0F5EE9A04FFFFFFFFFFFFFFFFFFFF739F|+288
/bin/sh: -h: not found
(shell) /bin/sh exited with non-zero return code: 127
telegram=|6E4401066052700405077ADA0060852F2F_0FCFAF6B9718020043A401847361F8444EFA0508000010882100007B018D1F0000271C0000051800005716000012140000C4110000780E0000960B0000440A0000090800003E05000052020000A037349B04FFFFFFFFFFFFFFFFFFFF1415|+299
/bin/sh: -h: not found
(shell) /bin/sh exited with non-zero return code: 127
```

Can you please point me into the right direction?
Sensors appear, but are not updated:
![image](https://github.com/wmbusmeters/wmbusmeters/assets/121175845/c4808963-0a2a-43a4-ba3f-9edd8c001444)

Thanks.

**Comment by DoktorPrase** [comment]
I have copied from the sourcecode the apator162.json into /homeassistant/wmbusmeters/etc/mqtt_discovery and the "suspicious secodn thing" is gone.

**Comment by weetmuts** [comment]
shell=shell=/usr/bin/mosquitto_pub
looks like one shell= too many? :-)

**Comment by DoktorPrase** [comment]
oh. my. Sorry for being stupid (but I have warned you :D ).

In the meantime I have restarted a couple times and since the MQTT discovery started working (I guess after coyping the apator162.json to the MQTT discovery folder) I deleted the mqtt section of the configuration.yaml - lo and behold, the sensors finally appeared, but they are there twice - one as a "sensor integration" and one as a "MQTT integration":
![image](https://github.com/wmbusmeters/wmbusmeters/assets/121175845/2a554119-c2c8-44be-bcf1-7656c2b1ccf1)
